public/app/features/explore/Logs/LogsTableWrap.tsx

```diff
@@ -55,8 +56,28 @@ export type FieldNameMeta = (InactiveFieldMeta | ActiveFieldMeta) & GenericMeta;
 type FieldName = string;
 type FieldNameMetaStore = Record<FieldName, FieldNameMeta>;
 
+function addJsonToDataFrame(dataFrame: DataFrame) {
+  const line = dataFrame.fields.find((field) => field.name === 'Line' || field.name === 'body');
+
+  if (line && line.values.length > 0 && line.values[0]) {
+    if (isLogLineJSON(line.values[0])) {
```

Comment:
If the first log is valid JSON, but subsequent are not, they will be formatted as JSON, so when users inspect things might be broken.
Similarly, if the first log line is not valid JSON, but subsequent logs are, they will not be formatted as JSON.